File: Week_01-数组-链表-堆栈-队列/917.仅仅反转字母.py
#
# @lc app=leetcode.cn id=917 lang=python
#
# [917] 仅仅反转字母
#

# @lc code=start

class Solution(object):
    def reverseOnlyLetters(self, s):
        """
        :type s: str
        :rtype: str
        """
        n = len(s)
        ans = list(s)  #把string初始化成字符数组
        left = 0
        right = n -1
        while left < right:
            while left < right and not ans[left].isalpha():
                left += 1
            while left < right and not ans[right].isalpha():
                right -= 1
            ans[left], ans[right] = ans[right], ans[left]
            left += 1
            right -= 1
        return ''.join(ans)#把数组中的字符合并成string
        
# @lc code=end

File: Week_01-数组-链表-堆栈-队列/917.仅仅反转字母.cpp
/*
 * @lc app=leetcode.cn id=917 lang=cpp
 *
 * [917] 仅仅反转字母
 */

// @lc code=start
class Solution {
public:
	// 判断是否是字符的方法 
    bool isAlpha (char c) {
        return (c >= 'A' && c <= 'Z') || (c >='a' && c <= 'z');
    }
    string reverseOnlyLetters(string S) {
        int left = 0, right = S.size() - 1;
        while (left < right) {
            if (!isAlpha(S[left])) {
                left++;
            } else if (!isAlpha(S[right])) {
                right--;
            } else {
                swap (S[left], S[right]);
                left++;
                right--;
            }
        }
        return S;
    }
};
// @lc code=end

